Lib/Parser/parser.c

```diff
@@ -0,0 +1,306 @@
+			break;
+
+		case CLEARSCREEN_CMD_ID:
+			s_cmd.ID = CLEARSCREEN_CMD_ID;
+			len = parser_readText(cmd,character,1);
+			for(i=0;i<len;i++) s_cmd.clearscreen.color[i] = character[i];
+			break;
+
+		case WAIT_CMD_ID:
+			s_cmd.ID = WAIT_CMD_ID;
+			s_cmd.wait.msecs = parser_readValue(cmd,1);
+			break;
+
+		case REPEAT_CMD_ID:
+			s_cmd.ID = REPEAT_CMD_ID;
+			s_cmd.repeat.amount = parser_readValue(cmd,1);
+			s_cmd.repeat.times = parser_readValue(cmd,2);
+			break;
+
+		case CIRCLE_CMD_ID:
+			s_cmd.ID = CIRCLE_CMD_ID;
+			s_cmd.circle.x = parser_readValue(cmd,1);
+			s_cmd.circle.y = parser_readValue(cmd,2);
+			s_cmd.circle.radius = parser_readValue(cmd,3);
+
+			len = parser_readText(cmd,character,4);
+			for(i=0;i<len;i++) s_cmd.circle.color[i] = character[i];
+			break;
+
+		case FIGURE_CMD_ID:
+			s_cmd.ID = FIGURE_CMD_ID;
+			s_cmd.figure.x1 = parser_readValue(cmd,1);
+			s_cmd.figure.y1 = parser_readValue(cmd,2);
+			s_cmd.figure.x2 = parser_readValue(cmd,3);
+			s_cmd.figure.y2 = parser_readValue(cmd,4);
+			s_cmd.figure.x3 = parser_readValue(cmd,5);
+			s_cmd.figure.y3 = parser_readValue(cmd,6);
+			s_cmd.figure.x4 = parser_readValue(cmd,7);
+			s_cmd.figure.y4 = parser_readValue(cmd,8);
+			s_cmd.figure.x5 = parser_readValue(cmd,9);
+			s_cmd.figure.y5 = parser_readValue(cmd,10);
+
+			len = parser_readText(cmd,character,11);
+			for(i=0;i<len;i++) s_cmd.figure.color[i] = character[i];
+			break;
+
+		case TOWER_CMD_ID:
+			s_cmd.ID = TOWER_CMD_ID;
+			s_cmd.tower.x = parser_readValue(cmd,1);
+			s_cmd.tower.y = parser_readValue(cmd,2);
+			s_cmd.tower.size = parser_readValue(cmd,3);
+
+			len = parser_readText(cmd,character,4);
+			for(i=0;i<len;i++) s_cmd.tower.color1[i] = character[i];
+			len = parser_readText(cmd,character,5);
+			for(i=0;i<len;i++) s_cmd.tower.color2[i] = character[i];
+			break;
+
+		default:
+			s_cmd.ID = UNKOWN_CMD_ID; //should never happen
+			break;
+	}
+	return s_cmd;	//returns struct
+}
+
+/**
+  * @brief  receives a buffer with all the incomming commands.
+  * 		Then divides them into single commands and parse
+  * 		them into the data structure
+  * @param 	buff buffer with all the insend commands
+  * @param  commands data structure type where parsed data is stored
+  * @param 	amount of commands incomming
+  * @retval error handling
+  */
+uint8_t parser_recData(char *buff, COMMAND *commands,uint8_t amount)
+{
+	int i = 0;
+	int j = 0;
+	static int k = 0;
+	uint8_t err = 0;
+	char cmd[150];
+	char *pbuf;
+	//array for easly testing types with strstr function
+	char types[AMOUNTOFCMDS][20]= {{"lijn"},
+								   {"rechthoek"},
+								   {"tekst"},
+								   {"bitmap"},
+								   {"clearscherm"},
+								   {"wacht"},
+								   {"herhaal"},
+								   {"cirkel"},
+								   {"figuur"},
+								   {"toren"}};
+
+	memset(cmd, 0, 150);										//reset cmd buffer
+	pbuf = &*buff;												//point to the start of the big data buffer
```

Comment:
Is dit niet hetzelfde als 'pbuf = buff' ?

Reply:
buff wijst naar de hele data dus alle commandos. pbuf is de pointer naar de volgende single command